src/boot/memory.c

```diff
@@ -408,6 +758,19 @@ void *load_segment_decompress(s32 segment, u8 *srcStart, u8 *srcEnd) {
             slidstart(compressed, dest);
 #elif MIO0
             decompress(compressed, dest);
+#elif LZ4
+            u32 lz4CompSize = *(u32 *) (compressed + 8);
+            extern int decompress_lz4_full_fast(const void *inbuf, int insize, void *outbuf, void* dmaCtx);
+            if (LIKELY(gIsConsole))
+                decompress_lz4_full_fast(compressed + DMA_ASYNC_HEADER_SIZE, lz4CompSize, dest, &asyncCtx);
+            else
+                lz4_unpack(compressed + DMA_ASYNC_HEADER_SIZE, lz4CompSize, dest, &asyncCtx);
+#elif LZ4T
```

Comment:
replace this section of code with its own small function call to shorten decompress function

Reply:
Implemented in header file for inlining